e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+        #     self,
+        #     "option",
+        #     "selected",
+        #     selected,
+        #     timeout=timeout,
+        #     is_checked=True,
+        # )
+
+    def expect_choice_groups(
+        self,
+        choice_groups: typing.Union[
+            # TODO; support patterns?
+            typing.List[str],
+            None,
+        ],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        """Expect choices to be in order"""
+        # Playwright doesn't like lists of size 0. Instead, use `None`
+        if choice_groups is not None and len(choice_groups) == 0:
+            choice_groups = None
+        if choice_groups is None:
+            playwright_expect(self.loc_choice_groups).to_have_count(0, timeout=timeout)
+            return
+
+        _RadioButtonCheckboxGroupBase._expect_locator_values_in_list(
+            self,
+            "optgroup",
+            "choice_groups",
+            choice_groups,
+            timeout=timeout,
+            key="label",
+        )
+
+    def expect_choice_labels(
+        self,
+        choice_labels: typing.Union[typing.List[PatternOrStr], None],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        # Playwright doesn't like lists of size 0. Instead, use `None`
+        if choice_labels is not None and len(choice_labels) == 0:
+            choice_labels = None
+        if choice_labels is None:
+            playwright_expect(self.loc_choices).to_have_count(0, timeout=timeout)
+            return
+        playwright_expect(
+            self.loc_container.locator(f"select#{self.id} option")
+        ).to_have_text(choice_labels, timeout=timeout)
+
+    # multiple: bool = False,
+    def expect_multiple(self, multiple: bool, *, timeout: Timeout = None) -> None:
+        expect_multiple(self.loc_select, multiple, timeout=timeout)
+
+    def expect_size_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(
+            self.loc_select,
+            "size",
+            value=value,
+            timeout=timeout,
+        )
+
+
+class InputSelect(_InputSelectBase):
+    # id: str,
+    # label: TagChildArg,
+    # choices: SelectChoicesArg,
+    # selected: Optional[Union[str, List[str]]] = None,
+    # multiple: bool = False,
+    # selectize: bool = False,
+    # width: Optional[str] = None,
+    # size: Optional[str] = None,
+    def __init__(self, page: Page, id: str) -> None:
+        super().__init__(
+            page,
+            id=id,
+            select_class="",
+        )
+
+    # selectize: bool = False,
+    def expect_selectize(self, selectize: bool, *, timeout: Timeout = None) -> None:
+        # class_=None if selectize else "form-select",
+        ex_selectize = playwright_expect(self.loc_select)
+        if selectize:
+            ex_selectize.not_to_have_class(re.compile("form-select"), timeout=timeout)
```

Comment:
These should be strings, not regexes

Reply:
They must be regexes, `to_have_class()` compares against the whole `class` value. Sad it does not check for whole class values

Reply:
4820eb9